wait fails assertion due to race condition

<h2>Pre-bug-report checklist</h2>

**1. This bug can be reproduced using YAML**
- [ ] Yes [👉 Please report a bug to the Argo Workflows GitHub 👈](https://github.com/argoproj/argo-workflows/issues/new/choose)
- [x] No

**2. This bug occurs when...**
- [ ] running Hera code without submitting to Argo (e.g. when exporting to YAML)
- [x] running Hera code and submitting to Argo

<h2>Bug report</h2>

**Describe the bug**
[Assertion](https://github.com/argoproj-labs/hera/blob/c7374e76e4ed07432f00d7bafeed8dd4c563b9f8/src/hera/workflows/workflow.py#L412) fails due to what I believe is a race condition with Argo starting the workflow. I am unclear if this is an issue with my OS and Argo cluster, or if this is a general bug. 

_Error log if applicable_:
```
AssertionError: workflow phase not defined for workflow status artifact_gc_status=None artifact_repository_ref=None compressed_nodes=None conditions=None estimated_duration=None finished_at=None message=None nodes=None offload_node_status_version=None outputs=None persistent_volume_claims=None phase=None progress=None resources_duration=None started_at=None stored_templates=None stored_workflow_template_spec=None synchronization=None
```

**To Reproduce**
Full Hera code to reproduce the bug:
```py
from hera.workflows import Steps, Workflow, script


@script()
def echo(message: str):
    print(message)


with Workflow(
    generate_name="single-script-",
    entrypoint="steps",
) as w:
    with Steps(name="steps"):
        echo(arguments={"message": "A"})

w.create()
w.wait() # offending line
```

**Expected behavior**
If I run the code below, then the script successfully waits for workflow completion. 
```py
w.create()
time.sleep(0.5) # added by me
w.wait() 
```

**Environment**
 - OS: Amazon Linux 2
 - Browser: N/A
 - Version of Hera: 5.5.2
 - Version of Argo: v3.4.8

**Additional context**
Add any other context about the problem here.

--- a/src/hera/workflows/workflow.py
+++ b/src/hera/workflows/workflow.py
@@ -405,6 +405,13 @@ class Workflow(
         assert self.namespace is not None, "workflow namespace not defined"
         assert self.name is not None, "workflow name not defined"
 
+        # here we use the sleep interval to wait for the workflow post creation. This is to address a potential
+        # race conditions such as:
+        # 1. Argo server says "workflow was accepted" but the workflow is not yet created
+        # 2. Hera wants to verify the status of the workflow, but it's not yet defined because it's not created
+        # 3. Argo finally creates the workflow
+        # 4. Hera throws an `AssertionError` because the phase assertion fails
+        time.sleep(poll_interval)
         wf = self.workflows_service.get_workflow(self.name, namespace=self.namespace)
         assert wf.metadata.name is not None, f"workflow name not defined for workflow {self.name}"
 

